moz-webgpu-cts/src/main.rs

```diff
@@ -125,6 +190,474 @@ fn run(cli: Cli) -> ExitCode {
     }
 
     match subcommand {
+        Subcommand::ProcessReports {
+            report_globs,
+            report_paths,
+            preset,
+        } => {
+            let report_globs = {
+                let mut found_glob_parse_err = false;
+                // TODO: I think these can just be args. directly?
+                let globs = report_globs
+                    .into_iter()
+                    .filter_map(|glob| match Glob::diagnosed(&glob) {
+                        Ok((glob, _diagnostics)) => Some(glob.into_owned().partition()),
+                        Err(diagnostics) => {
+                            found_glob_parse_err = true;
+                            let error_reports = diagnostics
+                                .into_iter()
+                                .filter(|diag| {
+                                    // N.B.: There should be at least one of these!
+                                    diag.severity()
+                                        .map_or(true, |sev| sev == miette::Severity::Error)
+                                })
+                                .map(Report::new_boxed);
+                            for report in error_reports {
+                                eprintln!("{report:?}");
+                            }
+                            todo!("render glob parse diagnostics in hard error")
+                        }
+                    })
+                    .collect::<Vec<_>>();
+
+                if found_glob_parse_err {
+                    log::error!("failed to parse one or more WPT report globs; bailing");
+                    return ExitCode::FAILURE;
+                }
+
+                globs
+            };
+
+            if report_paths.is_empty() && report_globs.is_empty() {
+                log::error!("no report paths specified, bailing");
+                return ExitCode::FAILURE;
+            }
+
+            let exec_report_paths = {
+                let mut found_glob_walk_err = false;
+                let files = report_globs
+                    .iter()
+                    .flat_map(|(base_path, glob)| {
+                        glob.walk(base_path)
+                            .filter_map(|entry| match entry {
+                                // TODO: warn when not ending in `wptreport.json`
```

Comment:
- [ ] Do it.

Reply:
Actually, broke out to https://github.com/ErichDonGubler/moz-webgpu-cts/issues/52, fixed up with b63cb52b16d0490c0812284ca660bf16a1da775f.